### PR TITLE
[WIP] Add initial filtering page and tagging extension infrastructure

### PR DESCRIPTION
**Pull request by cticenhour**
@GiudGiud once the initial tagging PR is in, this is the base needed to add your own keys (in `config.yml`) and have an HTML filter page available (with the base CSS and JS @friedmud provided). 

Current limitations with the site as-is: 

- The entries do not have clickable links and do not have context beyond the name. 
- The current filtering logic also never down-selects when multiple filters are added (i.e., clicking one tag gets a filtered list, but the second doesn't further filter. It might even add to the list if hits are found for that tag alone). 

We'll need to pursue improvements on these points within Derek's [Tagview repository](https://github.com/friedmud/tagview). For readability at the moment, I recommend that you first attempt to set up a given tag as a normal title string, instead of a simple ID as in the extension examples:

```
!tag name='High Temperature Gas Reactor (HTGR)' pairs=foo:baz bar:foobar
```

This will at least make the initial non-clickable entries usable against the current manual list of pages. 

Refs #254, idaholab/moose#25388

### CODE REVIEW

File: doc/content/resources/filter/assets/index-93b559a6.js
Comment (GiudGiud) [inline doc/content/resources/filter/assets/index-93b559a6.js:1]: how do I generate this?

**Comment by GiudGiud** [comment]
ah I should have used this PR. Oh well.

**Comment by cticenhour** [comment]
Yeah, thought I had pointed it out in the MOOSE PR. No big deal though. I generated that JS file using Derek's tagview repo (linked in the description). Reach out if you have an issue. I was able to create a conda environment, install node, and then build using npm.

**Comment by cticenhour** [comment]
The important data portion of that file is replaced by the extension, with the rest of it governing how the filtering is done.

**Comment by GiudGiud** [comment]
Closing this work moved to #283

**Pull request closed without merging**